src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,64 @@
+package duke.task;
+
+/**
+ * The Event class which represents an event task.
+ *
+ * @author Leong Jia Hao Daniel
+ */
+public class Event extends Task {
+
+    protected String time;
+
+    /**
+     * The constructor for Event.
+     *
+     * @param description The description of the event.
+     * @param time The time of the event.
+     */
+    public Event(String description, String time) {
+        super(description);
+        this.time = time;
+    }
+
+    /**
+     * Reads from the input file and returns a event based on the data
+     * in the file.
+     *
+     * @param string The string from the input file.
+     * @return A new event task describing the event.
+     */
+    public static Event parseFile(String string) {
+        String[] details = string.split(" \\| ");
+        Event event = new Event(details[2], details[3]);
+        if (details[1].equals("1")) {
+            event.markAsDone();
+        }
+        return event;
+    }
+
+    /**
+     * Overrides the toDataFormat() in task to return a String which
+     * is stored in the file.
+     *
+     * @return The task but formatted in the way it is meant to
+     *         be stored in the file.
+     */
+    @Override
+    public String toDataFormat() {
+        String completed = "0";
+        if (this.getStatusIcon().equals("X")) {
```

Comment:
I think the `this` keyword here might also be unnecessary because it is not shadowed by anything else. [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables) (This issue occurs a few more times throughout the code)

```suggestion
        if (getStatusIcon().equals("X")) {
```